### PR TITLE
#3 Add regex support for string based filters

### PR DESCRIPTION
**Pull request by sbcd90**

### CODE REVIEW

**Comment by MeneDev** [comment]
Thanks so much! Looks good at first sight, I'll look closer at it tomorrow 😊

File: dockproc/predicate.go [hunk 1]
```diff
@@ -91,7 +93,11 @@ type domainsPredicate struct {
 
 func (p domainsPredicate) Matches(ref dockref.Reference) bool {
 	for _, v := range p.domains {
-		if v == ref.Domain() {
+		if strings.HasPrefix(v, "/") && strings.HasSuffix(v, "/") {
```
Comment (MeneDev) [inline dockproc/predicate.go]: A `isRegex` function would increase code quality

**Review by MeneDev — COMMENTED** [review]
Good work!
Some remarks:
 * the check if the string represents RegEx is repeated (function would be nice, see [above](https://github.com/MeneDev/dockmoor/pull/17#pullrequestreview-164681883))
 * while the ticket only mentioned domain, name and tag, more string-based predicates have been added since the ticket creation (path, familiarName).

I'm fine with the current state if you are. Let me know if you want to close the ticket and PR or resolve the comments. Thanks again!

**Comment by sbcd90** [comment]
Hello @MeneDev , Thanks a lot for your review. I have taken care of both points now. Can you please have a look now?

**Comment by MeneDev** [comment]
Thanks again :)

**Pull request merged**